### PR TITLE
Parse and export using sabre/vobject 

### LINKED ISSUE #4: evaluate sabre-vobject
i came across to this library when investigating what is the `iCalcreator` library being used: https://github.com/fruux/sabre-vobject

few benefits i see:
- [composer](https://github.com/fruux/sabre-vobject#installation) compatible
- [active](https://github.com/fruux/sabre-vobject/pulse/monthly) development
- [versioned](https://github.com/iCalcreator/iCalcreator/issues/21) tags
- [php7](https://github.com/iCalcreator/iCalcreator/issues/25) compatible

altho latest version [requires php 5.5](https://github.com/fruux/sabre-vobject/blob/4.1.1/composer.json#L35) might be an issue considering dokuwiki itself requires php 5.3 as minimum version

but there's older 3.5 series which requires just [php 5.3](https://github.com/fruux/sabre-vobject/blob/3.5/composer.json#L8)

if you see the library fit, i see the integration as:
- create `composer.json`: `composer require sabre/vobject "^3.5"`
- add it to version control: `git add composer.json && git commit -a`
- install deps: `composer install`
- `require vendor/autoload.php` to initialize deps in code
- include `vendor/*` in release `.zip`

i actually use such method in [eventum](https://github.com/eventum/dokuwiki-plugin-eventum/) plugin. have look at the project `Makefile` of the zip build process

### PR DESCRIPTION
**Pull request by real-or-random**
Some testing would be very helpful. 

This fixes #4 and #11. 

The only drawback is that vobject requires PHP >= 5.5. (And the old version works with PHP 5.3 but not with PHP 7)

### CODE REVIEW

**Comment by glensc** [comment]
well. don't require specific version in composer.json then? then travis can pick best version that is able to resolve dependencies. for release zip, yes, probably just use php 5.5 build.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package_name = dokuwiki-plugin-icalevents
+version=`awk '/date/{print $$2}' plugin.info.txt`
+
+default: dist
+
+all:
+
+dist:
+	set -ex
```
Comment (glensc) [inline Makefile]: this is useless; each line in makefile is ran as in separate shell script. and `-e` behaviour is default in make and make will print out commands it runs anyway, so `-x` is also covered.

if you want to run in same shell script you should backslash escape the newlines:

```
set -xe; \
more stuff
```

**Comment by real-or-random** [comment]
Well, the API has changed between vobject versions 3 and 4. So I guess it's simpler to drop support for vobject3 (and thus PHP 5.3).

**Comment by real-or-random** [comment]
That worries me: https://www.dokuwiki.org/popularity
PHP 5.3 and 5.4 seem to be very common still.

**Comment by glensc** [comment]
I'm one of them :)

but I haven't updated this plugin for very long time, I probably need to send several PR's to get back functionality I need.

**Comment by real-or-random** [comment]
My plan is to go ahead and merge this, dropping support for PHP 5.3 and 5.4.
Support for PHP 5.4 has been dropped over 1.5 years ago. Even 5.5 is not supported anymore. So running these versions is dangerous anyway. https://secure.php.net/supported-versions.php 

I'll add a check to make sure I don't include the library in those old version... Let's not break entire DokuWiki installations.

File: syntax.php [hunk 2]
```diff
@@ -67,438 +62,22 @@ function connectTo($mode) {
...
 
-    function nowikiStart() {
-        return '<nowiki>' . $this->magicString();
-    }
 
-    function nowikiEnd() {
-        return $this->magicString() . '</nowiki>';
+// An 'require' ensures that older PHP versions do not even try to parse the actual code.
+if (version_compare(PHP_VERSION, '5.5.0') >= 0) {
+    require 'syntax-impl.php';
```
Comment (glensc) [inline syntax.php:77]: use `__DIR__` to require from absolute path, not from php include path (generic name might hit something)

also, compare `PHP_VERSION_ID` instead. `PHP_VERSION_ID` is available as of PHP 5.2.7.

Comment (real-or-random) [reply syntax.php:77]: Good catch, I'll change that. (But I don't think a new release is necesary.)

**Pull request merged**